linalg/sparsesmoothers.cpp

```diff
@@ -90,22 +130,35 @@ void DSmoother::Mult(const Vector &x, Vector &y) const
    {
       if (type == 0)
       {
-         oper->Jacobi(x, *p, *r, scale, use_abs_diag);
+         A.Jacobi(x, *p, *r, scale, use_abs_diag);
       }
       else if (type == 1)
       {
-         oper->Jacobi2(x, *p, *r, scale);
+         A.Jacobi2(x, *p, *r, scale);
       }
       else if (type == 2)
       {
-         oper->Jacobi3(x, *p, *r, scale);
+         A.Jacobi3(x, *p, *r, scale);
       }
       else
       {
-         mfem_error("DSmoother::Mult wrong type");
+         MFEM_ABORT("Invalid type.");
       }
       Swap<Vector*>(r, p);
    }
 }
 
+void DSmoother::Mult(const Vector &x, Vector &y) const
+{
+   Mult_(*oper, x, y);
+}
+
+void DSmoother::MultTranspose(const Vector &x, Vector &y) const
+{
+   EnsureTranspose();
```

Comment:
In the case of `!iterative_mode && type == 0 && iterations == 1`, when only `DiagScale` is called, can `oper` be used instead of copying the matrix entries to `At`?

Reply:
Yes, in fact this is true even for `type != 0`. Addressed in c95ab91de18f6a3f01cb7df1cbb6fa3ce477aab4.